packages/hash/api/src/graphql/resolvers/user/updateUser.ts

```diff
@@ -1,83 +1,98 @@
+  }
+
+  /** @todo: enable admins to have a shortname under 4 characters */
+  if (shortname.length < shortnameMinimumLength) {
+    throw new UserInputError("Shortname must be at least 4 characters long.");
+  }
+  if (shortname.length > shortnameMaximumLength) {
+    throw new UserInputError("Shortname cannot be longer than 24 characters");
+  }
+};
+
 export const updateUser: ResolverFn<
   Promise<UnresolvedGQLUser>,
   {},
   LoggedInGraphQLContext,
   MutationUpdateUserArgs
-> = async (_, { userEntityId, properties }, { dataSources, user }) =>
-  dataSources.db.transaction(async (client) => {
-    const { graphApi } = dataSources;
-    /** @todo: allow HASH admins to bypass this */
+> = async (
+  _,
+  { userEntityId, properties },
+  { dataSources: { graphApi }, user },
+) => {
+  /** @todo: run this mutation in a transaction */
 
-    if (userEntityId !== user.entityId) {
-      throw new ForbiddenError("You can only update your own user properties");
-    }
+  /** @todo: allow HASH admins to bypass this */
+  if (userEntityId !== user.entityId) {
+    throw new ForbiddenError("You can only update your own user properties");
+  }
 
-    const { shortname, preferredName, usingHow } = properties;
+  let updatedUser = user;
 
-    if (!shortname && !preferredName && !usingHow) {
-      throw new ApolloError(
-        "An updated 'shortname', 'preferredName' or 'usingHow' value  must be provided to update a user",
-        "NO_OP",
-      );
-    }
+  const { shortname, preferredName } = properties;
 
-    if (shortname) {
-      if (user.getShortname() === shortname && !preferredName) {
-        throw new ApolloError(
-          `User with entityId '${user.entityId}' already has the shortname '${shortname}'`,
-          "NO_OP",
-        );
-      }
+  if (shortname && user.getShortname() !== shortname) {
+    await validateShortname(graphApi, shortname);
 
-      await Account.validateShortname(client, shortname);
+    updatedUser = await updatedUser.updateShortname(graphApi, {
+      updatedByAccountId: user.accountId,
+      updatedShortname: shortname,
+    });
+  }
 
-      await user.updateShortname(graphApi, {
-        updatedByAccountId: user.accountId,
-        updatedShortname: shortname,
-      });
-    }
-
-    if (preferredName) {
-      if (user.getPreferredName() === preferredName) {
-        throw new ApolloError(
-          `User with entityId '${user.entityId}' already has the preferredName '${preferredName}'`,
-          "NO_OP",
-        );
-      }
-
-      if (!User.preferredNameIsValid(preferredName)) {
-        throw new ApolloError(
-          `The preferredName '${preferredName}' is invalid`,
-          "PREFERRED_NAME_INVALID",
-        );
-      }
-
-      await user.updatePreferredName(graphApi, {
-        updatedByAccountId: user.accountId,
-        updatedPreferredName: preferredName,
-      });
+  if (preferredName && user.getPreferredName() !== preferredName) {
```

Comment:
(follow up comment to the error never throwing)
I suppose the first part of the `if` check here is false for `""`, which means the validation never gets to run (right now validation is just `=== ""`)

Reply:
Correct